src/tests/inmemory-modeler/include/inmemory-modeler.h

```diff
@@ -0,0 +1,167 @@
+// Copyright 2007-2025, RTE (https://www.rte-france.com)
+// See AUTHORS.txt
+// SPDX-License-Identifier: MPL-2.0
+// This file is part of Antares-Simulator,
+// Adequacy and Performance assessment for interconnected energy networks.
+//
+// Antares_Simulator is free software: you can redistribute it and/or modify
+// it under the terms of the Mozilla Public Licence 2.0 as published by
+// the Mozilla Foundation, either version 2 of the License, or
+// (at your option) any later version.
+//
+// Antares_Simulator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// Mozilla Public Licence 2.0 for more details.
+//
+// You should have received a copy of the Mozilla Public Licence 2.0
+// along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+
+#pragma once
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "antares/expressions/visitors/EvaluationContext.h"
+#include "antares/expressions/visitors/TimeIndex.h"
+#include "antares/optimisation/linear-problem-data-impl/linearProblemData.h"
+#include "antares/optimisation/linear-problem-mpsolver-impl/linearProblem.h"
+#include "antares/solver/optim-model-filler/VariableDictionary.h"
+#include "antares/study/system-model/model.h"
+
+#include "inmemory-modeler.h"
+
+namespace Antares::Optimisation::LinearProblemDataImpl
+{
+class LinearProblemData;
+}
+
+namespace Antares::ModelerStudy::SystemModel
+{
+class Model;
+class Component;
+enum class TimeDependent : bool;
+enum class ScenarioDependent : bool;
+class Constraint;
+class Parameter;
+enum class ValueType;
+} // namespace Antares::ModelerStudy::SystemModel
+
+namespace Antares::Expressions::Visitors
+{
+enum class TimeIndex : unsigned int;
+enum class ParameterType : unsigned int;
+struct ParameterTypeAndValue;
+} // namespace Antares::Expressions::Visitors
+
+namespace Antares::Expressions::Nodes
+{
+class Node;
+class LiteralNode;
+class ParameterNode;
+class VariableNode;
+class MultiplicationNode;
+class NegationNode;
+} // namespace Antares::Expressions::Nodes
+
+namespace Antares::Optimization
+{
+class ComponentFiller;
+}
+
+namespace Antares::Optimisation::LinearProblemApi
+{
+class ILinearProblem;
+class FillContext;
+class LinearProblemFiller;
+} // namespace Antares::Optimisation::LinearProblemApi
```

Comment:
Those forward declarations are probably not needed, use declaration from headers when possible (special case: cyclic dependency)